AutoCollection/HttpDependencyParser.ts

```diff
@@ -116,17 +113,17 @@ class HttpDependencyParser extends RequestParser {
         if (typeof options === 'string') {
             if (options.indexOf("http://") === 0 || options.indexOf("https://") === 0) {
                 // protocol exists, parse normally
-                options = url.parse(options);
+                options = new url.URL(options);
             } else {
                 // protocol not found, insert http/https where appropriate
-                const parsed = url.parse(options);
-                if (parsed.host === "443") {
-                    options = url.parse("https://" + options);
+                const parsed = new url.URL("http://" + options);
```

Comment:
The protocol "http://" on this line is in order to make the new URL work - it requires protocol to parse url:
`new URL(input[, base])`
- input <string> The absolute or relative input URL to parse. If input is relative, then base is required. If input is absolute, the base is ignored.